### ISSUE TITLE
[wsl2] can't use ssh

### ISSUE THREAD

**Issue by transtone** [microsoft/WSL#4208]
in wsl2 when I use ssh to login a remote server, the connection closed immediately. 
I change the distro to v1, the connection can establish. 

when use git (with ssh-key) to push file to github,  it fail too. it says:
```bash
Received disconnect from 13.250.177.223 port 22:11: Bye Bye
Disconnected from 13.250.177.223 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by therealkenc** [comment]
> > is also problematic because it brings down the whole ssh key exchange (and by extension git) edifice.

> #4208

Might be #4199 related. But probably (?) not. There isn't enough in the OP to tell definitively. You can get that `fatal` for _all manner_ of reasons. Above you are on the _client_ side. Github doesn't care if your local IP address changed. On the other hand, you'd care a lot if Github's IP address kept changing and you couldn't do a DNS lookup on `github.com`.

Only way to tell conclusively would be set up your keys again and see if that fixes the problem. Which isn't really a WSL question, ever shifting client IP address or not. Not to diminish the other issue. [N.b. I retired from diving `ssh` problems a couple of years ago because `ssh` does work as advertised on both WSL1 and WSL2.]

**Comment by transtone** [comment]
because of this annoying bug, I decide not use the ssh client of wsl distro, but use OpenSSH in `C:\WINDOWS\System32\OpenSSH`. this do solve my problem.

**Comment by therealkenc** [comment]
> because of this annoying bug

Well let's be clear - no. No bug has been identified here. But happy to hear you are unblocked.

**Comment by transtone** [comment]
not a bug? Is this a feature?
if this is a feature, why saying [this](https://docs.microsoft.com/en-us/windows/wsl/wsl2-ux-changes#accessing-network-applications):<br>
`
In the initial builds of the WSL 2 preview, you will need to access any Linux server from Windows using the IP address of your Linux distro, and any Windows server from Linux using the IP address of your host machine. This is something that is temporary, and very high on our priority list to fix.
`

**Comment by therealkenc** [comment]
> if this is a feature, why saying this:

Because that paragraph has nothing to do with your OP. Git push to github works in WSL2 regardless of _whether_ "you will need to access any Linux server from Windows using the IP address of your Linux distro".

> No bug has been identified here

The word "identified" is important here. That _you_ cannot push to Github does not _identify_ a bug. Indeed, the scope for bugs has shrunken considerably in WSL2 (vs. WSL1) because you'd have to conclude you found something new and novel wrong with either (a) Hyper-V or (b) the Linux kernel. 

The reason for your "`fatal: Could not read from remote repository`" is, to be clear, neither of those things. Whatever the reason may be. And so the obv isn't lost: `git push` WSL2 to github works. That you've decided to use Windows OpenSSH instead, for whatever reasons, is fine also; that works too. 
The cap below is superfluous. If "[wsl] can't use ssh" were a _bug_, you'd know from the screeching.

![image](https://user-images.githubusercontent.com/7049700/59975037-709d3880-9568-11e9-9f8a-bf24b559e96a.png)

**Comment by transtone** [comment]
Thank you for let me know how to identify a bug in wsl. 
It must be something wrong with the network that makes ssh can't connect remote server, but I don't have the time to solve it. So I open a issue to let you guys know someone trouble about this, and how to avoid this. 
I'll try to solve this after wsl2 stable release, if it has the same problem.

**Comment by CamberLoid** [comment]
Encountered the same issue at WSL2, but after digging further I have a workaround here. 
Try to explicitly specify ciphers at /etc/ssh/ssh_config, or run `ssh -oCiphers=your_ciphers username@domain`. It works for me.

**Comment by KexyBiscuit** [comment]
@CamberLoid Open a new issue might help since this one has been closed for a long time.

**Comment by pdelaney-802** [comment]
I just upgraded to WSL2 along with latest Windows 10, 2020 and I cannot ssh to any of our servers now.  I have to go to the cmd or powershell it's a bummer.  I was excited about WSL2.

**Comment by thedevelopnik** [comment]
I have to add `ssh-add ~/.ssh/<name_of_private_key_file>` to my .zshrc file for it to work. It seems like every session the keys have to be added again. That's the bug to me, I would expect to only have to do that once ever.

**Comment by therealkenc** [comment]
> I have to add ssh-add ~/.ssh/<name_of_private_key_file> 2> /dev/null to my .zshrc file for it to work

That was mentioned in [message](https://github.com/microsoft/WSL/issues/4199#issuecomment-504226007) with open issue landing zones #4711 #4199 #4210 (depending on one's disposition).

**Comment by hiyyg** [comment]
I can not use ssh in wsl2. I have to change it back to wsl 1.

**Comment by robertzhangwenjie** [comment]
I also encountered this issue in wsl2, its a annoying bug.

**Comment by acmuller** [comment]
Just installed WSL2, and I also can't SSH to any of my servers. Rolling back to WSL1

**Comment by behnam-io** [comment]
I'm also facing this issue. Any fix since then?

**Comment by shimbalama** [comment]
same

**Comment by RoboticistAbdulrahman** [comment]
Hi all, i've found the solution here 
[(https://github.community/t/having-an-issue-pull-and-push-in-github/13198/2)]

- create a file  called config in folder ~/.ssh 
`cd ~/.ssh `
`nano config`
- paste the following 
`Host github.com
KexAlgorithms=ecdh-sha2-nistp521`
